crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -1436,22 +1455,33 @@ pub fn detect_fork(
     action: &Action,
 ) -> StateQueryResult<Option<(ActionHash, Signature)>> {
     let mut statement = txn.prepare(ACTION_HASH_BY_PREV)?;
-    statement
-        .query_row(
+    let items = statement
+        .query_map(
             named_params! {
                 ":prev_hash": action.prev_action(),
                 ":hash": action.to_hash(),
             },
-            |row| {
-                let hash: ActionHash = row.get("hash")?;
-                let action_blob: Vec<u8> = row.get("blob")?;
-                Ok((hash, action_blob))
+            |row| match row.get("hash") {
+                Ok(hash) => {
+                    let action_blob: Vec<u8> = row.get("blob")?;
+                    Ok(Some((hash, action_blob)))
+                }
+                Err(err) => match row.get::<_, WarrantHash>("hash") {
+                    Ok(_) => Ok(None),
+                    Err(_) => Err(err),
```

Comment:
I wonder if this is worth a code comment, took me a minute to decipher. So if we fail to read an action hash, then assume it's a warrant hash. If that fails then return the error for reading an action hash?

Either a comment, or maybe a custom error message in that case to say that it's not recognised as either of the expected hash types?